javatests/arcs/sdk/HandleHolderBaseTest.kt

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.sdk
+
+import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import kotlin.test.assertFailsWith
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(JUnit4::class)
+class HandleHolderBaseTest {
+
+  @Test
+  fun dispatcher_noHandles_asserts() {
+    val emptyHandleHolder = HandleHolderBase(
+      "TestParticle",
+      entitySpecs = mapOf("inSpec" to mock())
+    )
+
+    assertFailsWith(
+      IllegalStateException::class,
+      "No dispatcher available for a HandleHolder with no handles."
+    ) {
+      emptyHandleHolder.dispatcher
+    }
+  }
+
+  @Test
+  fun dispatcher_someHandles_getsFirstHandlesDispatcher() {
+    val handleHolder = HandleHolderBase(
+      "TestParticle",
+      entitySpecs = mapOf(
+        "readHandle" to mock(),
+        "writeHandle" to mock()
+      )
+    )
+    val firstHandle = mock<Handle> {
+      on { dispatcher } doReturn Dispatchers.Main
+    }
+    val secondHandle = mock<Handle> {
+      on { dispatcher } doReturn Dispatchers.Default
+    }
+    handleHolder.setHandle("readHandle", firstHandle)
+    handleHolder.setHandle("writeHandle", secondHandle)
+
+    val actual = handleHolder.dispatcher
+
+    assertThat(actual).isSameInstanceAs(Dispatchers.Main)
+  }
+
+  @Test
+  fun getEntitySpec_handleNameNotInSpec_throwsNoElemException() {
+    val handleHolder = HandleHolderBase(
+      "TestParticle",
+      entitySpecs = mapOf(
+        "readHandle" to mock(),
+        "writeHandle" to mock()
+      )
+    )
+
+    assertFailsWith(
+      NoSuchElementException::class,
+      "Particle TestParticle does not have a handle with name notInSpec."
+    ) {
+      handleHolder.getEntitySpecs("notInSpec")
+    }
+  }
+
+  @Test
+  fun getEntitySpec_handleNameInSpec_returnsSpec() {
+    val mock = mock<Set<EntitySpec<EntityBase>>>()
+    val handleHolder = HandleHolderBase(
+      "TestParticle",
+      entitySpecs = mapOf("readHandle" to mock)
+    )
+
+    assertThat(handleHolder.getEntitySpecs("readHandle")).isSameInstanceAs(mock)
+  }
+
+  @Test
+  fun setHandle_whenAlreadyInitialized_throwsNoElemException() {
```

Comment:
nit - it throws IllegalArgumentException, but i think for test name it is sufficient to just state `_throwsException`